### PR TITLE
First Version of User Auth System

### PR DESCRIPTION
**Pull request by XuechunHou**
Features Implemented:

1. Sign up
    - [X]  form field validation 

     - [x] basic UI implementation
2. Log in
     - [X] form field validation
     -  [X] basic UI implementation
     - [X] session cookie: a single login session last for 10 hrs 
3. Also created an ugly user profile page

### CODE REVIEW

File: web_server/social_distribution/settings.py [hunk 1]
```diff
@@ -43,15 +43,16 @@
     'posts.apps.PostsConfig',
     'comments.apps.CommentsConfig',
     'friendship.apps.FriendshipConfig',
-    'users.apps.UsersConfig'
+    'users.apps.UsersConfig',
+    'crispy_forms'
```
Comment (AustinGrey) [inline web_server/social_distribution/settings.py:47]: Crispy forms seems to be library you are using? Why? Do we have a place to document this so if the front end people try using something else we dont conflict?

**Review by AustinGrey — APPROVED** [review]
Please see my comment about crispy-forms

**Review by AustinGrey — CHANGES_REQUESTED** [review]
Actually I just tried to run this locally and couldn't because I didn't have crispy-forms. Do you mind adding a requirement.txt so that everyone can have the same modules installed in their venv? You should be able to make one from your venv by running pip --freeze

**Comment by XuechunHou** [comment]
`Django==2.2.9
django-clear-cache==0.3
**django-crispy-forms==1.8.1**
pytz==2019.3
sqlparse==0.3.0
`
django-crispy-forms is the only packages I donwloaded, it's a form styling framework..
pip install --upgrade django-crispy-forms

**Comment by AustinGrey** [comment]
Great, that's simple then. Please commit the requirements.txt file and push so that we have the dependency documented. We won't be able to use heroku without the dependencies listed in the requirements.txt

**Comment by AustinGrey** [comment]
Please see slack - it turns out everyone has been using different versions of Django. We should standardize. You have 2.2, 2 other team members were on 2.1, I had 3. Do you have any thoughts on that?

**Comment by AustinGrey** [comment]
As per slack, 2.2 is now the chosen version of Django. Thus this PR is in line and can continue.

**Pull request merged**